src/decoding/ringbuffer.rs

```diff
@@ -301,7 +301,7 @@ impl RingBuffer {
             unsafe {
                 let src = (
                     self.buf.as_ptr().cast_const().add(self.head + start),
-                    self.tail - self.head,
+                    self.tail - self.head - start,
```

Comment:
So this reads like tail - `offset` similar to the line above `self.head + start`.. Maybe it should be made an extra var to be more clear for future readers?